log4j-plugin-processor/src/main/java/org/apache/logging/log4j/plugin/processor/PluginProcessor.java

```diff
@@ -282,6 +315,25 @@ private String commonPrefix(final String str1, final String str2) {
         return str1.substring(0, minLength);
     }
 
+    private static boolean isServiceConsumerClassPresent() {
+        try {
+            Class.forName("aQute.bnd.annotation.spi.ServiceConsumer");
+            return true;
```

Comment:
This looks for `ServiceConsumer` on the plugin processor's classpath, while it should look for it on the application classpath.